polynote-frontend/polynote/ui/component/nb_config.ts

```diff
@@ -236,167 +227,207 @@ class NotebookConfigHandler extends UIMessageTarget {
+                excl: textbox(['exclusion'], 'Exclusion organization:name', value?.data.exclusion).change(() => {
                     excl.data.exclusion = excl.elements.excl.value
                 }),
                 remove: iconButton(['remove'], 'Remove', 'minus-circle-red', 'Remove').click(evt => {
                     this.exclusionContainer.removeChild(excl.row);
                     this.exclusions = this.exclusions.filter(e => e !== excl);
-                    if (this.exclusions.length === 0) this.addExcl(defaultExcl)
+                    if (this.exclusions.length === 0) this.addExcl()
                 }),
                 add: iconButton(['add'], 'Add', 'plus-circle', 'Add').click(evt => {
-                    this.addExcl(excl)
+                    this.addExcl()
                 }),
             },
             row: div(['exclusion-row', 'notebook-config-row'], []),
             data: {
-                exclusion: previous.data.exclusion,
+                exclusion: value?.data.exclusion ?? "",
             }
         };
 
         for (const el of Object.values(excl.elements)) {
             excl.row.appendChild(el);
         }
 
-        this.exclusionContainer.insertBefore(excl.row, previous.row);
-
+        this.exclusionContainer.appendChild(excl.row);
         this.exclusions.push(excl);
 
         return excl;
     }
 
-    addSparkConf(previous: SparkConf) {
+    addSparkConf(value?: SparkConf) {
         const conf = {
             elements: {
-                key: textbox(['spark-config-key'], 'key', previous.data.key).change(() => {
+                key: textbox(['spark-config-key'], 'key', value?.data.key).change(() => {
                     conf.data.key = conf.elements.key.value
                 }),
-                val: textbox(['spark-config-val'], 'val', previous.data.val).change(() => {
+                val: textbox(['spark-config-val'], 'val', value?.data.val).change(() => {
                     conf.data.val = conf.elements.val.value
                 }),
                 remove: iconButton(['remove'], 'Remove', 'minus-circle-red', 'Remove').click(evt => {
                     this.sparkConfigContainer.removeChild(conf.row);
                     this.sparkConfigs = this.sparkConfigs.filter(c => c !== conf);
-                    if (this.sparkConfigs.length === 0) this.addSparkConf(defaultSparkConf)
+                    if (this.sparkConfigs.length === 0) this.addSparkConf()
                 }),
                 add: iconButton(['add'], 'Add', 'plus-circle', 'Add').click(evt => {
-                    this.addSparkConf(conf)
+                    this.addSparkConf()
                 }),
             },
             row: div(['exclusion-row', 'notebook-config-row'], []),
             data: {
-                key: previous.data.key,
-                val: previous.data.val,
+                key: value?.data.key ?? "",
```

Comment:
Either you're going crazy with the new TS operator, or we've got a lot of broken stuff waiting to happen right now...

I mean, `newTsOperator?.goCrazy() ?? "a lot of broken stuff waiting to happen right now..."`

Reply:
I hope I understand the new `.?` properly 😛

Reply:
haha, well in this case I needed to use `value?.` because `value` can be undefined (but `previous` wasn't undefined so it didn't need it.)